go-controller/pkg/util/pod_annotation.go

```diff
@@ -693,3 +712,8 @@ func UnmarshalUDNOpenPortsAnnotation(annotations map[string]string) ([]*OpenPort
 	}
 	return result, nil
 }
+
+// Ensure the IP is valid IPv6 and falls within the `fe80::/10` LLA range
+func isIPv6LLA(ip net.IP) bool {
+	return ip != nil && ip.To4() == nil && ip[0] == 0xfe && ip[1]&0xc0 == 0x80
```

Comment:
There is wais at golang "netip" to check if it's link local 

https://pkg.go.dev/net/netip#Addr.IsLinkLocalUnicast

We should try to use those first before going into hex territory.

so something like 
 ```golang
if utilnet.IsIPv6(addr.IP) && addr.IP.IsLinkLocalUnicast()
```

Reply:
done thx